collector/monitors/update_global_index_monitor.go

```diff
@@ -0,0 +1,237 @@
+package monitors
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/lidofinance/terra-monitors/collector/config"
+	"github.com/lidofinance/terra-monitors/openapi/client"
+	"github.com/lidofinance/terra-monitors/openapi/client/transactions"
+	"github.com/lidofinance/terra-monitors/openapi/models"
+	"github.com/sirupsen/logrus"
+)
+
+type UpdateGlobalIndexTxsVariants int
+
+const (
+	NonUpdateGlobalIndexTX UpdateGlobalIndexTxsVariants = iota
+	SuccessfulUpdateGlobalIndexTX
+	FailedUpdateGlobalIndex
+)
+
+const UpdateGlobalIndexBase64Encoded = "eyJ1cGRhdGVfZ2xvYmFsX2luZGV4Ijp7fX0="
+
+const (
+	UpdateGlobalIndexSuccessfulTxSinceLastCheck Metric = "update_global_index_successful_tx_since_last_check"
+	UpdateGlobalIndexFailedTxSinceLastCheck     Metric = "update_global_index_failed_tx_since_last_check"
+	UpdateGlobalIndexGasWanted                  Metric = "update_global_index_gas_wanted"
+	UpdateGlobalIndexGasUsed                    Metric = "update_global_index_gas_used"
+	UpdateGlobalIndexUUSDFee                    Metric = "update_global_index_uusd_fee"
+)
+
+const threshold int = 10
+
+type UpdateGlobalIndexMonitor struct {
+	metrics          map[Metric]float64
+	ApiResponse      *models.GetTxListResult
+	ContractAddress  string
+	apiClient        *client.TerraLiteForTerra
+	logger           *logrus.Logger
+	lastMaxCheckedID int
+}
+
+func NewUpdateGlobalIndexMonitor(cfg config.CollectorConfig) UpdateGlobalIndexMonitor {
+	m := UpdateGlobalIndexMonitor{
+		metrics:         make(map[Metric]float64),
+		ContractAddress: cfg.UpdateGlobalIndexBotAddress,
+		apiClient:       cfg.GetTerraClient(),
+		logger:          cfg.Logger,
+	}
+
+	return m
+}
+
+func (m UpdateGlobalIndexMonitor) Name() string {
+	return "UpdateGlobalIndexMonitor"
+}
+
+func (m *UpdateGlobalIndexMonitor) InitMetrics() {
+	m.metrics[UpdateGlobalIndexSuccessfulTxSinceLastCheck] = 0
+	m.metrics[UpdateGlobalIndexGasWanted] = 0
+	m.metrics[UpdateGlobalIndexGasUsed] = 0
+	m.metrics[UpdateGlobalIndexUUSDFee] = 0
+	m.metrics[UpdateGlobalIndexFailedTxSinceLastCheck] = 0
+}
+
+func (m *UpdateGlobalIndexMonitor) Handler(ctx context.Context) error {
+	var offset *float64
+	var fetchedTxs int
+	var firstCheck bool
+	if m.lastMaxCheckedID == 0 {
+		firstCheck = true
+	}
+
+	iterations := 0
+	var maxProcessedID int
+	var alreadyProcessedFound bool
+	m.InitMetrics()
+	for iterations < threshold {
+		p := transactions.GetV1TxsParams{}
+		p.SetAccount(&m.ContractAddress)
+		p.SetContext(ctx)
+		p.SetOffset(offset)
+
+		resp, err := m.apiClient.Transactions.GetV1Txs(&p)
+		if err != nil {
+			return fmt.Errorf("failed to fetch transaction history for UpdateGlobalIndexBotContract account: %w", err)
+		}
+
+		maxProcessedID, alreadyProcessedFound = m.processTransactions(resp.Payload.Txs, m.lastMaxCheckedID)
+		fetchedTxs += len(resp.Payload.Txs)
+		maxProcessedID = maxInt(m.lastMaxCheckedID, maxProcessedID)
```

Comment:
Not an issue but I've noticed this line is meaningless cause ```maxProcessedID``` already greater or equal to ```m.lastMaxCheckedID``` (but never less) cause of L119

Reply:
An issue. There was actually a bug, witch lead to unexpected behavior and wrong m.lastMaxCheckedID in case we are fetching more than one txs batches. Fixed it.

Reply:
Oh, then nice that I noticed